packages/app-builder/src/components/Form/FormCheckbox.tsx

```diff
@@ -1,24 +1,32 @@
+import { unstable_useControl, useField } from '@conform-to/react';
 import { Checkbox } from 'ui-design-system';
 
-import { useFieldConfig } from './FormField';
+import { useFieldName } from './FormField';
+
+interface FormCheckboxProps
+  extends Omit<React.ComponentProps<typeof Checkbox>, 'checked'> {}
 
 export function FormCheckbox({
   children,
   onCheckedChange,
   ...rest
-}: Omit<React.ComponentProps<typeof Checkbox>, 'checked'>) {
-  const config = useFieldConfig();
+}: FormCheckboxProps) {
+  const name = useFieldName();
+  const [meta] = useField<boolean>(name);
+
+  const control = unstable_useControl(meta);
```

Comment:
sanity check: what should I make of "unstable_" ?

Reply:
Nothing 🤷🏽 

This is a common convention for pkg mainteners to tell the public API is unstable and may be changed with breaking changes in the future; still the hook is introduced for more than 4 months now, w\ any changes. This was created in order to fix issues with the initial attempt of handling form custom inputs `useInputControl`.

More info [here](https://github.com/edmundhung/conform/releases/tag/v1.0.2)